packages/consumption/src/modules/requests/itemProcessors/createAttribute/CreateAttributeRequestItemProcessor.ts

```diff
@@ -30,30 +34,66 @@ export class CreateAttributeRequestItemProcessor extends GenericRequestItemProce
                 );
             }
 
-            if (typeof recipient !== "undefined") {
+            return ValidationResult.error(
+                ConsumptionCoreErrors.requests.invalidRequestItem(
+                    "The owner of the provided IdentityAttribute for the `attribute` property can only be the address of the recipient or an empty string. The latter will default to the address of the recipient."
+                )
+            );
+        }
+
+        if (!(recipientIsAttributeOwner || senderIsAttributeOwner || ownerIsEmptyString)) {
+            return ValidationResult.error(
+                ConsumptionCoreErrors.requests.invalidRequestItem(
+                    "The owner of the provided RelationshipAttribute for the `attribute` property can only be the address of the sender, the address of the recipient or an empty string. The latter will default to the address of the recipient."
+                )
+            );
+        }
+
+        if (typeof recipient !== "undefined") {
```

Comment:
The `getRelationshipAttributesOfValueTypeToPeerWithGivenKeyAndOwner` method needs the `recipient` for its required `peer` input parameter. This method needs the `peer` because it checks the existing RelationshipAttributes. Other methods I have implemented do not need to know the `recipient` because they only ensure key uniqueness within the Request and do not take existing RelationshipAttributes into account.

If I have not understood you correctly, please let me know.